stan/math/rev/fun/sd.hpp

```diff
@@ -14,51 +14,88 @@ namespace stan {
-var sd(const T& m) {
-  check_nonzero_size("sd", "m", m);
-  if (m.size() == 1) {
-    return 0;
+template <typename T, require_eigen_st<is_var, T>* = nullptr>
+var sd(const T& x) {
+  using std::sqrt;
+  using T_vi = promote_scalar_t<vari*, T>;
+  using T_d = promote_scalar_t<double, T>;
+
+  check_nonzero_size("sd", "x", x);
+
+  if (x.size() == 1) {
+    return 0.0;
   }
 
-  return apply_vector_unary<T>::reduce(m, [](const auto& dtrs_map) {
-    using std::sqrt;
-    using T_map = std::decay_t<decltype(dtrs_map)>;
-    using T_vi = promote_scalar_t<vari*, T_map>;
-    using T_d = promote_scalar_t<double, T_map>;
-    vari** varis = ChainableStack::instance_->memalloc_.alloc_array<vari*>(
-        dtrs_map.size());
-    double* partials = ChainableStack::instance_->memalloc_.alloc_array<double>(
-        dtrs_map.size());
-    Eigen::Map<T_vi> varis_map(varis, dtrs_map.rows(), dtrs_map.cols());
-    Eigen::Map<T_d> partials_map(partials, dtrs_map.rows(), dtrs_map.cols());
+  vari** varis
+      = ChainableStack::instance_->memalloc_.alloc_array<vari*>(x.size());
+  double* partials
+      = ChainableStack::instance_->memalloc_.alloc_array<double>(x.size());
+  Eigen::Map<T_vi> varis_map(varis, x.rows(), x.cols());
+  Eigen::Map<T_d> partials_map(partials, x.rows(), x.cols());
 
-    varis_map = dtrs_map.vi();
-    T_d dtrs_val = varis_map.val();
-    double mean = dtrs_val.mean();
-    T_d diff = dtrs_val.array() - mean;
-    double sum_of_squares = diff.squaredNorm();
-    double size_m1 = dtrs_map.size() - 1;
-    double sd = sqrt(sum_of_squares / size_m1);
+  varis_map = x.vi();
+  T_d dtrs_val = x.val();
+  double mean = dtrs_val.mean();
+  T_d diff = dtrs_val.array() - mean;
+  double sum_of_squares = diff.squaredNorm();
+  double size_m1 = x.size() - 1;
+  double sd = sqrt(sum_of_squares / size_m1);
 
-    if (sum_of_squares < 1e-20) {
-      partials_map.fill(inv_sqrt(static_cast<double>(dtrs_map.size())));
-    } else {
-      partials_map = diff.array() / (sd * size_m1);
-    }
-    return var(new stored_gradient_vari(sd, dtrs_map.size(), varis, partials));
+  if (sum_of_squares < 1e-20) {
+    partials_map.fill(inv_sqrt(static_cast<double>(x.size())));
+  } else {
+    partials_map = diff.array() / (sd * size_m1);
+  }
+  return var(new stored_gradient_vari(sd, x.size(), varis, partials));
+}
+
+/**
+ * Return the sample standard deviation of the var_value matrix
+ *
+ * @tparam T Input type
+ * @param[in] x input matrix
+ * @return sample standard deviation of specified matrix
+ * @throw domain error  size is not greater than zero.
+ */
+template <typename T, require_var_matrix_t<T>* = nullptr>
+var sd(const T& x) {
+  check_nonzero_size("sd", "x", x);
+
+  if (x.size() == 1) {
+    return 0.0;
+  }
+
+  auto arena_diff = to_arena((x.val().array() - x.val().mean()).matrix());
+  double sum_of_squares = arena_diff.squaredNorm();
+  double sd = std::sqrt(sum_of_squares / (x.size() - 1));
```

Comment:
[optional] You could do the little loop thing here to make this faster but it's fine as is

Reply:
> You have to fix up the docs

Do you mean the doxygen docs or the function reference docs (second definitely needs updated and I wouldn't doubt the first lol)?

Reply:
Oh doxygen my bad

Reply:
@SteveBronder I don't know what was going on with the docs. I changed all the variables to be named `x` to get them to work. That was a real hairtugger